pkg/agent/datamodel/helper.go

```diff
@@ -93,3 +93,24 @@ func IndentString(original string, spaces int) string {
 	}
 	return out.String()
 }
+
+func processContainerImage(downloadURL string) (string, error) {
+	// example URL "downloadURL": "mcr.microsoft.com/oss/kubernetes/autoscaler/addon-resizer:*",
+	// getting the data between the last / and the last :
+	parts := strings.Split(downloadURL, "/")
+	if len(parts) == 0 || len(parts[len(parts)-1]) == 0 {
+		return "", fmt.Errorf("container image component URL is not in the expected format: %s", downloadURL)
+	}
+	lastPart := parts[len(parts)-1]
+	component := strings.TrimSuffix(lastPart, ":*")
+	return component, nil
+}
+
+func processDownloadFile(downloadURL string) (string, error) {
+	// example URL "downloadURL": "https://acs-mirror.azureedge.net/cni-plugins/v*/binaries",
+	parts := strings.Split(downloadURL, "/")
+	if len(parts) == 0 || len(parts) != 6 {
```

Comment:
can you please remind me why this split will always by 6? could this change in the future? having a hardcoded number in the API doesn't seem best but I want to understand the reasoning behind this to see if its unavoidable.

Reply:
https://acs-mirror.azureedge.net/cni-plugins/v*/binaries when split by "/" makes 6 indices 

[https:, "", acs-mirror.azureedge.net, cni-plugins, v*, binaries]

so I agree its not perfect, all the urls do follow this same pattern. If that was to change, we would need to adjust this function

Reply:
maybe we could utilize this instead? https://pkg.go.dev/net/url

Reply:
I still had to utilize some string slicing, but maybe I was able to use url.Parse() to make it a bit cleaner